### PR TITLE
Update LDAP API to lookup addl. attributes

### PR DESCRIPTION
**Pull request by donatello**
- With this change, if additional LDAP attributes are specified in the
configuration, the server will fetch them when a user's DN is looked up
via `LookupUsername` (renamed here from `LookupDN`) and `LookupUserDN`.
The return types of these APIs are changed.

- Bump up to v3 as this causes API breakage.

- Update CI actions versions

### CODE REVIEW

**Comment by donatello** [comment]
After merge, need to tag as `v3.0.0`

**Comment by donatello** [comment]
@harshavardhana need to update repo config about the required CI check (updated to go 1.22 here)

**Review by zveinn — APPROVED** [review]

**Comment by donatello** [comment]
Needed for https://github.com/minio/console/pull/3349 and https://github.com/minio/minio/pull/19758

File: ldap/ldap.go [hunk 1]
```diff
@@ -68,6 +76,11 @@ type Config struct {
 	UserDNSearchBaseDistNames []BaseDNInfo
 	UserDNSearchFilter        string
 
+	// Additional attributes to fetch from the user DN search.
+	UserDNAttributes string
+	// this is a computed value from UserDNAttributes
+	UserDNAttributesList []string
```
Comment (vadmeste) [inline ldap/ldap.go:82]: maybe s/UserDNAttributesList/userDNAttributesList ?

**Review by vadmeste — COMMENTED** [review]
One comment, LGTM otherwise

File: ldap/ldap.go (hunk 1 above)
Comment (donatello) [reply ldap/ldap.go:82]: Need this to be exported

Comment (harshavardhana) [reply ldap/ldap.go:82]: are these immutable values @donatello ?

Comment (donatello) [reply ldap/ldap.go:82]: Yes immutable

Comment (donatello) [reply ldap/ldap.go]: ok this is fixed now.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**